### PR TITLE
chore: adjust export strategy

### PR DESCRIPTION
**Pull request by FrankEssenberger**
Adjust the existing ADR as a consequence on our discussion.

### CODE REVIEW

File: knowledge-base/adr/0028-public-api-extraction.md [hunk 1]
```diff
@@ -52,21 +56,35 @@ Hence, we propose the following approach:
      This removes the internal API form the `d.ts` files.
   2. Maintain minimal named exports in the root `index.ts` pointing to the objects of the public API.
      This creates also minimal module exports in the JavaScript usecase.
+  3. We create a `internal` export as we did for the [v4 case](https://github.com/SAP/cloud-sdk-js/tree/v1.28.0/packages/core) which can be imported via `@sap-cloud-sdk/odata-v2/internal`.
+     This internal package contains all exports.
```
Comment (deekshas8) [inline knowledge-base/adr/0028-public-api-extraction.md:60]: `internal` is a folder right? The word `package` is a little confusing here.

File: knowledge-base/adr/0028-public-api-extraction.md [hunk 2]
```diff
@@ -52,21 +56,35 @@ Hence, we propose the following approach:
...
 However, the double maintenance makes better check rules possible.
 We plan to implement the following checks:
 
 - Have a check to avoid any `*` exports in the root `index.ts`.
-- Have a check to enforce a single `index.ts` on root level specifying all public exports.
+- Use an automatic tool like [barrelsby](https://github.com/bencoveney/barrelsby#readme) to create `index.ts` exporting everything to the `internal` folder.
 - Have a check to enforce exposed object from `d.ts` match the named exports of the root `index.ts`.
   - A missing value in the `index.ts` denotes: You have exported something in the code but not added it to the relevant barrel or missed the `@internal`.
   - A missing value in the `d.ts` denotes: You have violated the API contract and marked a previously exported object as `@internal`.
+- In the integration and E2E tests we enforce root level imports via eslint rule.
```
Comment (deekshas8) [inline knowledge-base/adr/0028-public-api-extraction.md:71]: Clarification: Should the vdm e2e tests also have only root level imports? As there are a couple of imports from `dist` in the template apps as well. If not, then we could use with the internal, e.g. `@sap-cloud-sdk/odata-v2/internal`

File: knowledge-base/adr/0028-public-api-extraction.md (hunk 1 above)
Comment (FrankEssenberger) [reply knowledge-base/adr/0028-public-api-extraction.md:60]: yes

File: knowledge-base/adr/0028-public-api-extraction.md (hunk 2 above)
Comment (FrankEssenberger) [reply knowledge-base/adr/0028-public-api-extraction.md:71]: I think the client side so the call to the backend-app should not have any internal imports. The backend side does not really matter.

File: knowledge-base/adr/0028-public-api-extraction.md [hunk 3]
```diff
@@ -49,24 +53,38 @@ Hence, we propose the following approach:
...
 - [Optiona] Have a check to enforce TsDoc on all exported objects.
   Either a full doc if part of the public API or at least `@internal` as minimal value.
   The [eslint-plugin-tsdoc](https://github.com/microsoft/tsdoc/issues/209) does not have a `require` rule.
   The [eslint-plugin-jsdoc](https://github.com/gajus/eslint-plugin-jsdoc) does have it but does not recognize many TypeScript object as `interface`.
 
+## Details on Package
```
Comment (jjtang1985) [inline knowledge-base/adr/0028-public-api-extraction.md:77]:
```suggestion
## Internal Packages
```

File: knowledge-base/adr/0028-public-api-extraction.md [hunk 4]
```diff
@@ -27,8 +27,12 @@ This was convenient for us, but we had no distinction between:
 
 Our new strategy should provide the following options:
 
-- `Exported Public API`: This is a cautious decision and consumer can rely on stable contract for minor versions. This API is reachable via the root level import.
-- `Exported Internal API`: Exported for technical reasons but not meant to be used by consumer so no stable contract. This API is not accessible without [dirty tricks](https://www.goodreads.com/quotes/7714510-c-protects-against-accident-not-against-fraud).
+- `Exported Public API`: This is a cautious decision and consumer can rely on stable contract for minor versions.
+  It contains only objects where we see (re)use potential for customers.
+  This API is reachable via the root level import e.g. `@sap-cloud-sdk/odata-v2`.
+- `Exported Internal API`: Exported for technical reasons but not meant to be used by consumer.
+  We keep this API stable for path versions.
```
Comment (jjtang1985) [inline knowledge-base/adr/0028-public-api-extraction.md:34]: Do we really want to have this contract?
I would not make any promises if possible and only tells please use internal APIs with caution.

**Review by jjtang1985 — APPROVED** [review]
LGTM.
I left some minor comments.

File: knowledge-base/adr/0028-public-api-extraction.md (hunk 4 above)
Comment (FrankEssenberger) [reply knowledge-base/adr/0028-public-api-extraction.md:34]: Yes I would like to keep for patch releases the internal API stable. I mean we can always do a minor release if needed.

**Pull request merged**